multimodal/src/autogluon/multimodal/learners/base_learner.py

```diff
@@ -0,0 +1,3514 @@
+                    classes=self._classes,
+                    init_scratch=self._init_scratch,
+                )
+
+    @property
+    def path(self):
+        if self._matcher:
+            return self._matcher.path
+        else:
+            return self._save_path
+
+    @property
+    def label(self):
+        if self._matcher:
+            return self._matcher.label
+        else:
+            return self._label_column
+
+    @property
+    def query(self):
+        if self._matcher:
+            return self._matcher.query
+        else:
+            warnings.warn("Matcher is not used. No query columns are available.", UserWarning)
+            return None
+
+    @property
+    def response(self):
+        if self._matcher:
+            return self._matcher.response
+        else:
+            warnings.warn("Matcher is not used. No response columns are available.", UserWarning)
+            return None
+
+    @property
+    def match_label(self):
+        if self._matcher:
+            return self._matcher.match_label
+        else:
+            warnings.warn("Matcher is not used. No match_label is available.", UserWarning)
+            return None
+
+    @property
+    def problem_type(self):
+        return self._problem_type
+
+    @property
+    def problem_property(self):
+        if self._problem_type is None:
+            return None
+        else:
+            return PROBLEM_TYPES_REG.get(self._problem_type)
+
+    @property
+    def column_types(self):
+        if self._matcher:
+            return self._matcher.column_types
+        else:
+            return self._column_types
+
+    # This func is required by the abstract trainer of TabularPredictor.
+    def set_verbosity(self, verbosity: int):
+        """Set the verbosity level of the log.
+
+        Parameters
+        ----------
+        verbosity
+            The verbosity level.
+
+            0 --> only errors
+            1 --> only warnings and critical print statements
+            2 --> key print statements which should be shown by default
+            3 --> more-detailed printing
+            4 --> everything
+
+        """
+        self._verbosity = verbosity
+        set_logger_verbosity(verbosity)
+        transformers.logging.set_verbosity(verbosity2loglevel(verbosity))
+
+    def _setup_save_path(self, save_path, fit_called):
+        return setup_save_path(
+            resume=self._resume,
+            old_save_path=self._save_path,
+            proposed_save_path=save_path,
+            raise_if_exist=True,
+            warn_if_exist=False,
+            fit_called=fit_called,
+        )
+
+    def _setup_train_tuning_data(
+        self,
+        train_data: Union[pd.DataFrame, str],
+        tuning_data: Optional[Union[pd.DataFrame, str]],
+        holdout_frac: Optional[float],
+        seed: Optional[int],
+        max_num_tuning_data: Optional[int],
+        **kwargs,
+    ):
+        if self._problem_type == OBJECT_DETECTION:
```

Comment:
I think this function is overridden by object detection learner and this if statement is not necessary